Update readme (#27)
Sage-Bionetworks/dccmonitor: README.md

## Installation

dccmonitor uses the reticulate package with the [Synapse Python Client](https://github.com/Sage-Bionetworks/synapsePythonClient). See the [reticulate documentation](https://rstudio.github.io/reticulate/#python-version) for more information on setting up reticulate to work with your local Python environment. Additionally, see the [Synapse Python Client](https://github.com/Sage-Bionetworks/synapsePythonClient) for installation instructions.

dccmonitor can be installed via devtools:
``` R
devtools::install_github("Sage-Bionetworks/dccmonitor")
```

## Shiny Application

Using the dccmonitor shiny application requires that the user have a Synapse account, and the user needs access to the specific Synapse folder of interest for monitoring (see Customization for details).

After installing the package, the following will run the application locally:

``` R
library(dccmonitor)

> shiny::runApp()

Reviewer: ```suggestion
run_app()
```

`shiny::runApp()` only works if you're in a directory with an app.R file or server.R/ui.R, i.e. if you've cloned the repo. `run_app()` should work regardless.

Author: That's a good point. I didn't think of the differences between cloning and installing. I had originally put `run_app()`, but when I did it, I realized that the config setting is skipped (since it's not running from app.R) and changed it back to `shiny::runApp()`. I think I'll add instructions for both ways.

Reviewer: If you want inspiration, @kelshmo has updated the language about this for dccvalidator here: https://github.com/Sage-Bionetworks/dccvalidator/pull/242

Author: Nice! I'll check it out.

- `teams`: Synapse team ID. This is used to verify that the user is a member of the team and, therefore, has access to the folder containing the metadata and documentation files of interest.
- `consortium_fileview`: Synapse file view ID. A Synapse file view that shows all the files in the `parent` folder (this is the folder that dccvalidator uploads metadata and documentation files to), along with their annotations. The file view should include the columns: id, name, createdOn, createdBy, modifiedOn, currentVersion, study, metadataType, species, assay. Note that the following file-specific annotations are required for dccmonitor to function properly:
    - documentation: study
    - manifest: study, metadataType = manifest
    - individual metadata: study, species, metadataType = individual
    - biospecimen metadata: study, species, metadataType = biospecimen
    - assay metadata: study, species, assay, metadataType = assay

> Simply set a configuration for the application in the config.yml and change "default" in app.R to your configuration name before running.

Reviewer: ```suggestion
Create a configuration for the application in the config.yml and change "default" in app.R to your configuration name before running.
```

I'd like to avoid "simply", as this might be unfamiliar to some folks and they might not find it simple at first.

Author: That's a great point!

# dccmonitor

> This package is intended to assist Sage Bionetworks data curators to check the status of metadata and documentation files uploaded via the [dccvalidator](https://sage-bionetworks.github.io/dccvalidator/) shiny application. dccmonitor is primarily a shiny application, with the ability to use functions for gathering validation information using the package functions only, if desired.

Author: The intention is to describe the function of the package. I added that bit because it's not required to use the shiny app at all. If someone really loved using R in the terminal and printing validation results to the console, they could do exactly that, using the package functions to gather the validation information. The package has a shiny application, but the package is not only a shiny application, if that makes sense.

Would it be more clear to say, "The dccmonitor package contains functions to gather validation information and a shiny application to view the results", instead?

> This package is intended to assist Sage Bionetworks data curators to check the status of metadata and documentation files uploaded via the [dccvalidator](https://sage-bionetworks.github.io/dccvalidator/) shiny application. dccmonitor is primarily a shiny application, with the ability to use functions for gathering validation information using the package functions only, if desired.

Reviewer 2: got it! I would say "The dccmonitor package contains functions to gather validation information. Optionally, a shiny application is also available to view the results."

**Note:** The application takes time to fully populate the validation information and currently does not have a progress indicator. The application will initialize the study boxes after checking for team membership and gathering the studies represented in the `consortium_fileview` (see Customization for details on this file view). After this, the application will populate the study boxes after the validation checks have run for all represented studies.

## Requirements

> dccmonitor uses the reticulate package with the [Synapse Python Client](https://github.com/Sage-Bionetworks/synapsePythonClient). See the [reticulate documentation](https://rstudio.github.io/reticulate/#python-version) for more information on setting up reticulate to work with your local Python environment. Additionally, see the [Synapse Python Client](https://github.com/Sage-Bionetworks/synapsePythonClient) for installation instructions.

Author: The Synapse Python Client does not have those instructions, but the reticulate package says how to set a specific python environment. I can add a sentence saying that the Synapse client needs to be in the same python environment that reticulate is pointed to.

## Installation and Use

dccmonitor can be installed and used in two ways: package installation, and cloning the repository.

### Package Installation

dccmonitor can be installed via devtools:

``` R
devtools::install_github("Sage-Bionetworks/dccmonitor")
```

To run the Shiny application, there needs to be a config.yml file in the working directory with the required options (see [Customization](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#customization)). After creating the config.yml file, the application can be run with:

``` R
library(dccmonitor)
Sys.setenv(R_CONFIG_ACTIVE = "default") # Replace "default" with the configuration name if not using default.
run_app()
```

### Cloning Repository

[Clone the repository](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) to a local directory. Customize the config.yml file, as needed (see [Customization](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#customization)).

> If your working directory is the application directory, you can change the app.R file to have your configuration name, if not using the default configuration. Then the following will start the app:

Reviewer 2: I suggest providing an example of how to change the configuration from default.

Author: I'm not sure what to put as an example for that since it's changing a single word in a file. The app.R file has a comment in it that says where to change the default. Do you have an example so I can better understand what you are looking for?

Reviewer 2: I took a look at the app.R file and I see your comment which is clear enough.

Author: Great!

``` R
renv::restore() # Update project package library
shiny::runApp()
```

## Customization

Customizing the application is done via the config.yml file and the specific file validation checks.

### config.yml

> A configuration file is required for the application behave correctly. Create a configuration for the application called config.yml (see [this repo's config.yml](https://github.com/Sage-Bionetworks/dccmonitor/blob/master/config.yml) for an example) in your working directory if you have installed the package, or alter the config.yml file in the cloned repository. If not using default as your configuration, set the active configuration as described in the instructions for your choice of installation, [package](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#package-installation) or [cloned repository](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#cloning-repository). Alternatively, if you have forked the repository and added a new set of configuration options to the config.yml file, you can open a Pull Request to see about adding the new options.

Reviewer 2: grammatical revision
```suggestion
A configuration file is required for the application to behave correctly. Create a configuration for the application called config.yml (see [this repo's config.yml](https://github.com/Sage-Bionetworks/dccmonitor/blob/master/config.yml) for an example) in your working directory if you have installed the package, or alter the config.yml file if you have cloned the repository. If not using default as your configuration, set the active configuration as described in the instructions for your choice of installation, [package](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#package-installation) or [cloned repository](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#cloning-repository). Alternatively, if you have forked the repository and added a new set of configuration options to the config.yml file, you can open a Pull Request to see about adding the new options.
```

This section is a bit difficult to follow - it may make sense to nest configuration under each installation option.

Author: This section is a summarized repeat of the installation instructions specific to the config file. I repeated them in a summarized format since that section is about that file (and in case someone skipped over the instructions altogether). I could remove everything in that paragraph after the first sentence and add about about referring to the instructions.

> A configuration file, called config.yml, is required for the application behave correctly. See the installation instructions above for information on where the file should be located and how to set the environment configuration.

Would this be easier to understand?

Reviewer 2: I think it is better as is, linking to each relevant section. Thanks!

> A configuration file is required for the application to behave correctly. Create a configuration for the application called config.yml (see [this repo's config.yml](https://github.com/Sage-Bionetworks/dccmonitor/blob/master/config.yml) for an example) in your working directory if you have installed the package, or alter the config.yml file if you have cloned the repository. If not using default as your configuration, set the active configuration as described in the instructions for your choice of installation, [package](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#package-installation) or [cloned repository](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#cloning-repository). Alternatively, if you have forked the repository and added a new set of configuration options to the config.yml file, you can open a Pull Request to see about adding the new options.

Reviewer 2: grammatical suggestion
```suggestion
A configuration file is required for the application to behave correctly. Create a configuration for the application called config.yml (see [this repo's config.yml](https://github.com/Sage-Bionetworks/dccmonitor/blob/master/config.yml) for an example) in your working directory if you have installed the package, or alter the config.yml file if you have cloned the repository. If you are not using default as your configuration, set the active configuration as described in the instructions for your choice of installation, [package](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#package-installation) or [cloned repository](https://github.com/Sage-Bionetworks/dccmonitor/tree/update-readme#cloning-repository). Alternatively, if you have forked the repository and added a new set of configuration options to the config.yml file, you can open a Pull Request to see about adding the new options.
```
